controlplane/internal/controller/openshiftassistedcontrolplane_controller_test.go

```diff
@@ -506,8 +517,9 @@ var _ = Describe("Upgrade scenarios", func() {
 		Expect(k8sClient.Update(ctx, openshiftAssistedControlPlane)).To(Succeed())
 
 		mockUpgradeFactory.EXPECT().NewUpgrader(gomock.Any()).Return(mockUpgrader, nil)
-		mockUpgrader.EXPECT().IsUpgradeInProgress(gomock.Any()).Return(false, nil)
+		mockUpgrader.EXPECT().IsUpgradeInProgress(gomock.Any()).Return(true, nil) // should this be successfully starting upgrade?
```

Comment:
Hey @rccrdpccl should this test `should handle upgrade with repository override` return true that upgrade is in progress? I wasn't sure

Reply:
According to the current behaviour, this should start an upgrade either way:
* if UpgradeInProgress, but DesiredVersion is not updated, we should trigger an update
* if Upgrade not in progress, and desiredVersion is not updated, we should still trigger an update.

I think the logic is fair, but would this work with CVO? Can we update the version if an upgrade is already running?

Reply:
Good question, I can verify that!